kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java

```diff
@@ -246,28 +238,127 @@ public static void assertPartitionOffsetsEqual(List<PartitionOffset> a, List<Par
         valueDeserializerClassName, new Properties(), validateContents);
   }
 
+  /**
+   * Retries a given set of assertions, if not all of them succeed, using a pre-defined exponential
+   * backoff strategy.
+   *
+   * @param assertions A set of assertions which are expected to throw if not all of them succeed.
+   */
   public static void testWithRetry(Runnable assertions) {
-    testWithRetry(assertions, DEFAULT_EXP_BACKOFF_RETRIES, DEFAULT_INITIAL_BACKOFF);
+    testWithRetry(assertions, DEFAULT_EXP_BACKOFF_RETRIES, null, DEFAULT_RETRY_INTERVAL, true);
   }
 
-  public static void testWithRetry(Runnable assertions, int numRetries, Duration initialBackoff) {
-    Duration backoff = initialBackoff;
+  private static void testWithRetry(
+      Runnable assertions,
+      int numRetries,
+      Duration timeout,
+      Duration initialRetryInterval,
+      boolean isExponentialBackoff) {
+    long sleepNanos = initialRetryInterval.toNanos();
+    long start = System.nanoTime();
+    long elapsed;
+    long maxElapsed = timeout != null ? timeout.toNanos() : Long.MAX_VALUE;
     for (int i = 0;; i++) {
       try {
         assertions.run();
         return;
+      } catch (NoRetryException e) {
```

Comment:
Do we use this `NoRetryException` anywhere? If we don't, I would get rid of it. One less dependency.

Reply:
We do not, and it's not part of the [Kafka 2.7.0 API](https://kafka.apache.org/27/javadoc/index.html?overview-summary.html), but I was afraid that some components we query might throw it.

Still, what you're suggesting is exactly what I wanted to do, so consider it done.